app/src/main/java/com/trueedu/project/data/RealPriceManager.kt

```diff
@@ -92,6 +92,12 @@ class RealPriceManager @Inject constructor(
         // 데이터 추가
         requests.clear()
         requests.addAll(codesRequested)
+
+        val topName = requestStack.lastOrNull()?.first
+        if (topName == name) {
+            // 이미 존재하는 name 이면 replace
+            requestStack.removeLast()
+        }
         requestStack.add(name to codesRequested)
```

Comment:
동일한게 들어있으면 넣다 뺄게 아니라, 아예 안 넣는건 어떻습니까?
```suggestion
        if (topName != name) {
            requestStack.add(name to codesRequested)
        }
```

Reply:
remove 해서 기존 걸 빼고 다시 넣는 겁니다. (replace)
새로 넣는 거면 기존 거 위에 쌓는 거고요 (push)

Reply:
name 이 키 이고, `codesRequested` 가 value 인  맵/dictionary 같은건가요?  그럼 ok

Reply:
네 그렇습니다. name 은 일종의 태그이고 실제 데이터는 `codesRequested` 입니다. ㄱㅅ